app/controllers/TransferAgreementController.scala

```diff
@@ -43,9 +38,22 @@ class TransferAgreementController @Inject()(val controllerComponents: SecurityCo
         .verifying("All records must be open", b => b)
     )(TransferAgreementData.apply)(TransferAgreementData.unapply)
   )
+  private val options: Seq[(String, String)] = Seq("Yes" -> "true", "No" -> "false")
+  implicit val language: Lang = langs.availables.head
+  private val addTransferAgreementClient = graphqlConfiguration.getClient[AddTransferAgreement.Data, AddTransferAgreement.Variables]()
+
+  def transferAgreement(consignmentId: UUID): Action[AnyContent] = secureAction.async { implicit request: Request[AnyContent] =>
+    val consignmentStatusService = new ConsignmentStatusService(graphqlConfiguration)
 
-  def transferAgreement(consignmentId: UUID): Action[AnyContent] = consignmentExists(consignmentId) { implicit request: Request[AnyContent] =>
-    Ok(views.html.transferAgreement(consignmentId, transferAgreementForm, options))
+    for (consignmentStatus <- consignmentStatusService.consignmentStatus(consignmentId, request.token.bearerAccessToken)) yield {
+      val transferAgreementStatus: String = consignmentStatus.flatMap(_.transferAgreement).getOrElse("")
```

Comment:
Rather than `getOrElse`, which implies that there's a good default value, you could leave this as an `Option[String]` and then do a case match rather than the `if` statement:

```
transferAgreementStatus match {
  case Some("Completed") => // Return the "already completed" version
  case _ => // Return the TA form
}
```